src/lib/gadgets/range-check.ts

```diff
@@ -12,6 +13,21 @@ export {
   twoLMask,
 };
 
+/**
+ * Asserts that x is in the range [0, 2^32)
+ */
+function rangeCheck32(x: Field) {
```

Comment:
I am happy to refactor this as well and move all the helpers (mainly `rangeCheckHelper`) out of `Field` entirely if we think we should

Reply:
yes we should! btw `rangeCheckHelper` is a bad API that should've never been exposed in this way. It returns a value that fits within `16*n` bits but is otherwise unconstrained. Only by asserting it to be equal to something else, it becomes a range check. But our current docs frame it as it would do provable reduction modulo `16n`